app/api/bookyt/api/bookings.rb

```diff
@@ -29,6 +30,7 @@ class Bookings < Grape::API
               debit_account: debit_account,
               comments: params[:comments],
             }
+            attributes.merge!(reference: Invoice.find(params[:invoice_id])) if params[:invoice_id]
```

Comment:
I just added a validation into the API params to handle this. Do you mind checking again?